tests/e2e/utils/create-url.js

```diff
@@ -0,0 +1,42 @@
+/**
+ * This is an enhanced/fixed version of `@wordpress/e2e-utils.createURL`,
+ * the original one has not so usable API.
+ * The PR to upstream nad tests are pending at
+ * https://github.com/WordPress/gutenberg/pull/34051
+ */
+/**
+ * External dependencies
+ */
+import { createURL as originalCreateUrl } from '@woocommerce/e2e-utils'; // eslint-disable-line import/named
+const WP_BASE_URL = originalCreateUrl( '' );
+
+/**
+ * Creates new URL by parsing given WPPath and query params, relative to the WP base.
+ *
+ * Given `query` params are merged with the ones already present in the path.
+ * Any overlapping ones will be overwritten.
+ *
+ * @example
+ *
+ * ```js
+ * createURL( '/foo?a=b&bar=f', { bar: 'baz', fiz: 'a/b/c' } );
+ * // "http://localhost:8889/foo?a=b&bar=baz&fiz=a%2Fb%2Fc"
+ * ```
+ *
+ * @param {string}              [WPPath=''] String to be serialized as pathname.
+ * @param {string|Object|Array} [query]     Query parameters in any format supported by {@link https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams/URLSearchParams | URLSearchParams}.
+ * @return {string} String which represents full URL.
+ */
+export function createURL( WPPath = '', query = '' ) {
+	const url = new URL( WPPath, WP_BASE_URL );
```

Comment:
❓ 
The `WPPath` is passed to `new URL` as the first parameter, so it could accept absolute or relative URL. Consider that the meaning of this variable has been changed, I think the variable name and its JSDoc might need to be adjusted as well.

Reply:
Initially, I wanted to limit the number of changes to the minimum in the upstream PR, but you're with, the changed behavior should be reflected in the name.

I wonder what would be better. Formally that's "url string": https://url.spec.whatwg.org/#url-writing : relative or absolute URL string with almost everything optional.
- `url` - to match MDN definition of `new URL( url )`, and somehow match `url string` in spec, but give a premise of being not a string but something stringifiable to URL string?

Then we could have something like:
Creates new absolute (full) URL string by parsing given url and `query` relative to the WP base.
```js
/**
 * Creates new absolute (full) URL string by parsing given
 * `url` and `query` relative to the WP base.
 *
 * @param {string}              [url=''] A valid URL string, usually a relative path, but could contain as well contain query, fragment or be absolute.
 * @param {string|Object|Array} [query]  Query parameters in any format supported by {@link https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams/URLSearchParams | URLSearchParams}.
 * @return {string} String which represents the full, absolute URL.
 */
export function createURL( url = '', query = '' ) {
	const fullUrl = new URL( url, WP_BASE_URL );
```

Reply:
I think it's good and in this way, it can match the absolute URL usage of the below as well. 👍 

https://github.com/woocommerce/google-listings-and-ads/blob/2878bcd89b9cbfc122c74e230ba1ac5873f03e7d/tests/e2e/utils/visit-gla-page.js#L30